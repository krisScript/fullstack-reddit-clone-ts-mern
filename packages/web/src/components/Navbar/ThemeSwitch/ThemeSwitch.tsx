import React, { FC, ReactNode } from 'react';
import { Switch } from 'antd';
interface ThemeSwitchProps {
  toggleTheme: () => void;
}
const ThemeSwitch: FC<ThemeSwitchProps> = ({ toggleTheme }) => {
  return (
    <Switch
      checkedChildren="Light"
      unCheckedChildren="Dark"
      defaultChecked
      onChange={toggleTheme}
      data-testid="theme-switch"
    />
  );
};

export default ThemeSwitch;
