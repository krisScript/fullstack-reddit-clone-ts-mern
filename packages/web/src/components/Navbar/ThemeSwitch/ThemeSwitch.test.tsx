import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import ThemeSwitch from './ThemeSwitch';
describe('<ThemeSwitch />', () => {
  const toggleTheme = jest.fn();
  const { container, getByTestId } = render(
    <ThemeSwitch toggleTheme={toggleTheme} />,
  );
  beforeEach(() => {
    const themeSwitch = getByTestId('theme-switch');
    fireEvent.click(themeSwitch);
  });
  afterEach(cleanup);

  it('snapshot', () => {
    expect(container).toMatchSnapshot();
  });
  it('should change input value', async () => {
    expect(toggleTheme).toHaveBeenCalledTimes(1);
  });
});
