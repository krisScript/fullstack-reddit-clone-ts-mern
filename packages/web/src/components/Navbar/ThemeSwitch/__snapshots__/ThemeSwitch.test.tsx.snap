// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemeSwitch /> snapshot 1`] = `
<div>
  <button
    ant-click-animating="false"
    aria-checked="false"
    class="ant-switch"
    data-testid="theme-switch"
    role="switch"
    type="button"
  >
    <span
      class="ant-switch-inner"
    >
      Dark
    </span>
  </button>
</div>
`;
